### ISSUE TITLE
Problems when extending to new datasets

### ISSUE THREAD

**Issue by ZeyuPing** [ManipTrans/ManipTrans#27]
Hi! Thansks for your great work! 
I'm trying to extend Maniptrans with new datasets. However, the difference on coordinate makes it a little hard. 
Could you please provide more details about the reference coordinate of object pose, etc.
Thanks!!!

**Comment by KailinLi** [comment]
Hi, thank you very much for your interest in our work!

I’m not completely sure which “reference coordinate of object pose” you are referring to.  
• If you mean the world coordinate system in simulation, the origin is set at the center of the table and the negative Z-axis points in the gravity direction.  
• For the OakInk-V2 dataset, we keep exactly the same coordinate system as in the original OakInk release.

Hope this helps— let me know if you need any further details!

**Comment by ZeyuPing** [comment]
> Hi, thank you very much for your interest in our work!
> 
> I’m not completely sure which “reference coordinate of object pose” you are referring to. • If you mean the world coordinate system in simulation, the origin is set at the center of the table and the negative Z-axis points in the gravity direction. • For the OakInk-V2 dataset, we keep exactly the same coordinate system as in the original OakInk release.
> 
> Hope this helps— let me know if you need any further details!

Thank you for your patience, and apologies for the unclear question. In the ManipData class, the loaded obj_pose and wrist_pose should be both defined in Isaac’s world coordinate system, correct? e.g. their reference coordinate system.
Related codes:
```
data = {
    "data_path": self.data_pathes[idx],
    "obj_id": "-1",  # ? placeholder
    "obj_verts": rs_verts_obj,
    "obj_urdf_path": os.path.join(self.data_dir, "102_obj.urdf"),
    "obj_trajectory": torch.tensor(
        np.stack(obj_pose[::self.skip].cpu()), 
        device=self.device, 
        dtype=torch.float
    ),
    "scene_objs": [],  # ? placeholder
    "wrist_pos": wrist_pos,
    "wrist_rot": wrist_rot,
    "mano_joints": mano_joints,
}
```
Thanks!

**Comment by KailinLi** [comment]
Hi, thanks for the follow-up.

• `obj_pose` and `wrist_pose` always have to be expressed in the same coordinate frame; otherwise, the hand–object interaction will be wrong.  
• In the snippet you showed, both poses follow the world frame convention used by the original QuasiSim implementation.  

If things still look incorrect, I suggest visualizing the hand and object together—seeing them in 3D usually makes it easier to spot any mistaken conversions. Coordinate transforms can indeed be a bit tedious.

Hope this helps!

**Comment by ZeyuPing** [comment]
Thanks a lot!!!!!

**Comment by KailinLi** [comment]
Closing this issue as it has been resolved. Feel free to reopen it if needed. Thanks!
